### PR TITLE
One Video: Added banner support for Dynamic Ad Placement (DAP)

### PR DESCRIPTION
**Pull request by adam-browning**
## Type of change

- [X ] Feature

## Description of change
Added 'banner' support for Dynamic Ad Placement (DAP) feature in the oneVideo adapter.
Now when mediaTypes: {banner: { sizes: [300, 250]}} is passed the oneVideo adapter will send a bid-request as well for the DAP response.

- test parameters for validating bids
1. adUnit mediaTypes must be set to banner.
2. bidder params must remain video and you must pass display: 1
3. DAP will only respond if the Video SSP Exchange has DAP enabled.
4. You can use this test Exchange by entering pubId: 'HBExchangeDAP'
5. The DAP unit can only render within an iframe - use GAM to do so as provided in http://prebid.org/dev-docs/examples/basic-example.html
```
{
  bidder: 'oneVideo',
  params: {
                    video: {
                        playerWidth: 300,
                        playerHeight: 250,
                        mimes: ['video/mp4', 'application/javascript'],
                        api: [1, 2],
                        display: 1,
                    },
                    pubId: 'YOUR_PUBLISHER_ID'
                }
}
```

Be sure to test the integration with your ad server using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

## Other information

Hi @DeepthiNeeladri,
Could you please review my pull request.
I have tested this locally and it is working now as expected:
You can review this live demo using a hosted version of the fixed adaptor with prebid.js v3.19.0
https://cdn.vidible.tv/prod/2020-05/06/5eb2714cb7eac40001500f52_v1.html

### Here is a summary of the changes I have made to https://github.com/prebid/Prebid.js/blob/master/modules/oneVideoBidAdapter.js

1. 11: supportedMediaTypes: ['video', 'banner'], Added "banner" type to support DAP flow
2. 85: Deleted bidResponse.mediaType = 'video', Moved logic below to row 90
3. new line 90: Added If this is DAP request set response mediaType as "banner"
4. 91-95: If this is DAP response, append the adm into bidResponse.ad not only bidResponse.vastXml
5. 96: Wrapped the outstream renderer setter only if this is Not a DAP response.

### CODE REVIEW

File: modules/oneVideoBidAdapter.js [hunk 1]
```diff
@@ -82,18 +93,30 @@ export const spec = {
...
       netRevenue: true,
       adUnitCode: bidRequest.adUnitCode
     };
+
+    if (bidRequest.mediaTypes.banner) {
+      bidResponse.mediaType = 'banner'
+    } else {
+      bidResponse.mediaType = 'video'
+    }
+
```
Comment (DeepthiNeeladri) [inline modules/oneVideoBidAdapter.js:107]: suggestion : bidResponse.mediaType = (bidRequest.mediaTypes.banner) ? 'banner' : 'video';

**Comment by adam-browning** [comment]
Hi @DeepthiNeeladri,
I have updated the following:
- video and banner validations
- tests
- examples .md file
thank you!
Adam

**Comment by adam-browning** [comment]
Hi @DeepthiNeeladri,
I've made the changes and awaiting your approval.
Thank you!

**Comment by adam-browning** [comment]
Hello @aleksatr,
Sorry to bother, but do you believe you will have time to review this PR anytime soon?
Much appreciated 
Adam

File: package-lock.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "prebid.js",
-  "version": "3.18.0-pre",
+  "version": "3.19.0-pre",
```
Comment (aleksatr) [inline package-lock.json:3]: You should not modify `package-lock.json` in this PR

File: modules/oneVideoBidAdapter.js [hunk 3]
```diff
@@ -25,6 +25,17 @@ export const spec = {
...
     }
 
+    // Prevend DAP Outstream validation
+    if (bid.mediaTypes.video) {
+      if (bid.mediaTypes.video.context === 'outstream' && bid.params.video.display === 1) {
+        return false;
+      }
+    }
+    // Banner DAP validation
+    if (bid.mediaTypes.banner && (typeof bid.params.video.display === 'undefined' || !bid.params.video.display)) {
```
Comment (aleksatr) [inline modules/oneVideoBidAdapter.js:35]:
```
typeof bid.params.video.display === 'undefined' ||
```

^^ This check is redundant, `!bid.params.video.display` would evaluate to `true` in that case as well.

**Comment by aleksatr** [comment]
hey @adam-browning , sorry for the delay, please check comments I added. Btw I was not able to get a test bid with params you provided, even on the test page you provided (was connected to CA node via VPN).

File: modules/oneVideoBidAdapter.js (hunk 3 above)
Comment (adam-browning) [reply modules/oneVideoBidAdapter.js]: understood 👍

**Comment by adam-browning** [comment]
Hi @aleksatr, 
Thank you for reviewing and I have pushed a new version excluding the redundant validation.
Regarding the test bidder params, please use the following:

`  var adUnits = [
    {
      code: 'banner-1',
      mediaTypes: {
        banner: {
          sizes: [300, 250]
        }
      },
      bids: [
        {
          bidder: 'oneVideo',
          params: {
            video: {
              playerWidth: 300,
              playerHeight: 250,
              mimes: ['video/mp4', 'application/javascript'],
              display: 1
            },
            site: {
              id: 1,
              page: 'https://verizonmedia.com',
              referrer: 'https://verizonmedia.com'
            },
            pubId: 'HBExchangeDAP'
          }
        }
      ]
    }
]`
* The test creative responds only with in US west coast and with a site/referrer override.
This example is also shown in the .md file (last example)

Much appreciated!
Adam

**Comment by aleksatr** [comment]
hey @adam-browning , I see that it's working properly, if you can resolve the package-lock.json comment, we can go ahead and merge this.

https://github.com/prebid/Prebid.js/pull/5199#discussion_r425122211

If you want to update package-lock.json you should create another PR for that, don't update it as part of adapter PR.

**Comment by adam-browning** [comment]
Hi @aleksatr, I just resolve the package-lock.json comment. Is that enough?

**Comment by aleksatr** [comment]
@adam-browning I still see that package-lock.json is modified?

**Comment by adam-browning** [comment]
Sorry @aleksatr, 
The current  package-lock.json was updated to "version": "3.19.0-pre", when I pulled the branch last time. should I manually update it to "version": "3.18.0-pre" ?

**Comment by aleksatr** [comment]
@adam-browning , I think there's a misunderstanding here :) 
What I'm trying to communicate is that you should not change package-lock.json at all, you should revert all changes you made to this file.

**Comment by adam-browning** [comment]
Thanks for clarifying @aleksatr 
After reverting the package-lock.json is not showing in PR now

**Comment by aleksatr** [comment]
LGTM

**Comment by adam-browning** [comment]
Thank you @aleksatr for your time and patience!
Much appreciated!

**Comment by aleksatr** [comment]
@adam-browning np, have a nice day! :)

**Pull request merged**